src/include/alloc.h

```diff
@@ -37,13 +37,28 @@ static ncclResult_t ncclCalloc(T** ptr, size_t nelem) {
   return ncclSuccess;
 }
 
+struct __attribute__ ((aligned(64))) allocationTracker {
+  uint64_t totalAlloc;
+  uint64_t totalAllocSize;
+  uint64_t align[6];
+};
+static_assert(sizeof(struct allocationTracker) == 64, "allocationTracker must be size of 64 bytes");
+#define MAX_ALLOC_TRACK_NGPU 32
+extern struct allocationTracker allocTracker[];
+
 template <typename T>
 static ncclResult_t ncclCudaCalloc(T** ptr, size_t nelem, bool isFineGrain = false) {
   if (isFineGrain)
     CUDACHECK(hipExtMallocWithFlags((void**)ptr, nelem*sizeof(T), hipDeviceMallocFinegrained));
   else
     CUDACHECK(hipMalloc(ptr, nelem*sizeof(T)));
   CUDACHECK(hipMemset(*ptr, 0, nelem*sizeof(T)));
+  int dev;
+  CUDACHECK(hipGetDevice(&dev));
+  if (dev < MAX_ALLOC_TRACK_NGPU) {
+    __atomic_fetch_add(&allocTracker[dev].totalAlloc, 1, __ATOMIC_SEQ_CST);
```

Comment:
The number of allocs is being counted, but I don't think it's being reported anywhere.

Reply:
The information is not useful outside RCCL. Thus I decided not to print it.

Reply:
But why even track then?  How do you check that value?

Reply:
We can check with gdb. Anyway, original thought was to use counter to track hipMalloc/hipFree pair so we can detect leak. But this requires create a wrapper and change all location of hipFree. We can discuss offline to see if this is worth doing.